lib/vdsm/virt/vm.py

```diff
@@ -1182,6 +1182,29 @@ def _load_corrected_timeout(self, base, doubler=20, load=None):
             load = len(self.cif.vmContainer)
         return base * (doubler + load) / doubler
 
+    def _disk_preparation_timeout(self):
+        """
+        Return approximate timeout for disk preparation.
+
+        Some disk devices require modifying and reloading udev rules.
+        Currently, udev rules are reloaded for each such device individually.
+        This function returns a wild-guess timeout value based on the number
+        of such devices.
+
+        :returns: The timeout in seconds (int).
+        """
+        disk_params = vmdevices.common.storage_device_params_from_domain_xml(
+            self.id, self._domain, self._md_desc, self.log)
+        n_disks_to_appropriate = len(
+            [True for drive in disk_params
+             if type(drive) is dict and 'GUID' in drive and
+             not drive.get('managed')]
+        )
+        per_disk_time = config.getfloat(
+            'vars', 'migration_listener_timeout_per_disk_preparation'
```

Comment:
migration_prepare_disk_timeout?

Will avoid the acrobatics you have to do to fetch the value in 80 lines.

Reply:
Unfortunately event that is too long.  But it is a good idea to use a shorter name anyway, what do you think about the new one?